loki/transformations/loop_blocking.py

```diff
@@ -0,0 +1,121 @@
+# (C) Copyright 2018- ECMWF.
+# This software is licensed under the terms of the Apache Licence Version 2.0
+# which can be obtained at http://www.apache.org/licenses/LICENSE-2.0.
+# In applying this licence, ECMWF does not waive the privileges and immunities
+# granted to it by virtue of its status as an intergovernmental organisation
+# nor does it submit to any jurisdiction.
+
+
+from loki import Subroutine, ir, FindNodes, Assignment, Sum, Product, Loop, LoopRange, IntLiteral, \
+    Quotient, InlineCall, DeferredTypeSymbol, Transformer, Array, Variable, RangeIndex, Scalar, \
+    SubstituteExpressions, FindVariables
+
+
+def split_loop(routine: Subroutine, loop: ir.Loop, block_size: int):
+    """
+    Assumes that loop is contiguous in blocking variable
+    into a blocked loop in those variables. Does not block
+    the loop data
```

Comment:
We are following the [Numpy style](https://www.sphinx-doc.org/en/master/usage/extensions/example_numpy.html) for Sphinx documentation, i.e., start with a 1-line summary of what the routine does, add additional description/limitation in a new paragraph below, and document type and purpose of parameters at the end (important: anything after the `Parameters` and `Returns` section are ignored when generating the HTML docs).